src/satosa/backends/saml2.py

```diff
@@ -232,13 +243,59 @@ def construct_requested_authn_context(self, entity_id):
 
         return authn_context
 
-    def authn_request(self, context, entity_id):
+    def _get_requested_attributes(self, requested_attributes):
+        if not requested_attributes:
+            return
+
+        attrs = self.converter.from_internal_filter(
+            self.attribute_profile, requested_attributes
+        )
+        requested_attrs = []
+        for attr in attrs:
```

Comment:
I think this can be also written like the following to avoid the nested loops and the appending to list:
```python
from itertools import product

attrs_req_attrs_product = product(attrs, self.requested_attributes)

requested_attrs = [
    dict(friendly_name=attr, required=req_attr['required'])
    for (attr, req_attr) in attrs_req_attrs_product 
    if req_attr['friendly_name'] == attr
]
```